homeassistant/components/unifi_direct/device_tracker.py

```diff
@@ -37,104 +33,43 @@
 def get_scanner(hass: HomeAssistant, config: ConfigType) -> UnifiDeviceScanner | None:
     """Validate the configuration and return a Unifi direct scanner."""
     scanner = UnifiDeviceScanner(config[DOMAIN])
-    if not scanner.connected:
-        return None
-    return scanner
+    return scanner if scanner.success_init else None
 
 
 class UnifiDeviceScanner(DeviceScanner):
     """Class which queries Unifi wireless access point."""
 
     def __init__(self, config):
         """Initialize the scanner."""
-        self.host = config[CONF_HOST]
-        self.username = config[CONF_USERNAME]
-        self.password = config[CONF_PASSWORD]
-        self.port = config[CONF_PORT]
-        self.ssh = None
-        self.connected = False
-        self.last_results = {}
-        self._connect()
+        self.clients = {}
+        self.ap = UniFiAP(
+            target=config[CONF_HOST],
+            username=config[CONF_USERNAME],
+            password=config[CONF_PASSWORD],
+            port=config[CONF_PORT],
+        )
+        self.success_init = self._update_clients()
```

Comment:
Can we move this side effect out of the constructor? I know it was there before but its a bad practice.

Reply:
sorry I'm not sure I understand...
should I remove this var? and instead on line 36 use:
`return scanner if scanner._update_clients() else None`
(would then need to make the _update_client not "private")

Reply:
I think that's the best yes